libs/blocks/quiz/utils.js

```diff
@@ -266,10 +266,29 @@ const findMatchForSelections = (results, selections) => {
 
   const userSelectionLen = selections.primary.length; // 1 - lr
 
-  if (userSelectionLen <= 1) {
+  // Case 1 - when no user selection is matched default result is returned.
+  if (userSelectionLen < 1) {
     return defaultResult;
   }
 
+  let singleProductResults;
+
+  // Case 2 - when exactly one product is matched and it is not grouped with '&' in between.
+  if (userSelectionLen === 1) {
+    // eslint-disable-next-line max-len
+    singleProductResults = results.find((destination) => {
+      if (destination.result.indexOf('&') === -1 && destination.result.includes(selections.primary[0])) {
+        return destination;
+      }
+    });
+  }
+
+  if (singleProductResults) {
```

Comment:
Hi @narcis-radu! Thanks for having a look. when `userSelectionLen` is not 1, that usecase is handled by `Case 1` and `Case 3. Line#270 and Line #292. Line #277-289 is not executed when `userSelectionLen` is not 1.

Reply:
Yes, that's correct and that's my suggestion as well: group the code into a single block so that not even L#286 gets evaluated, something like
```js
// Case 2 - when exactly one product is matched and it is not grouped with '&' in between.
if (userSelectionLen === 1) {
  const singleProductResults = results.find((destination) => {
    if (destination.result.indexOf('&') === -1 
      && destination.result.includes(selections.primary[0])) {
      return destination;
    }
  });

  if (singleProductResults) {
    recommendations.push(singleProductResults);
    return recommendations;
  }
}
```

Reply:
Ah..  I see what u r saying and that makes sense. however, Erich and I were talking offline about if we can skip this Case 2 entirely and combine both Case 2 & 3 into the compound product logic. Might update shortly after I validate all the usecases (until we have UT for this entire suite).

Reply:
Thanks @echampio-at-adobe ! Adjusted the code with that updated logic.

Reply:
@narcis-radu since the code block u referred to is removed now, I am merging this PR for now. Also, we will be bringing in UT for this entire quiz functionality soon which will give us more confidence while making critical logic changes like this. Thanks for your review.